services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/UploadPartCopyRequestIterable.java

```diff
@@ -70,12 +85,23 @@ public UploadPartCopyRequest next() {
                                                                partNumber,
                                                                uploadId,
                                                                range);
+            updateProgressListener();
             partNumber++;
             offset += partSize;
             remainingBytes -= partSize;
             return uploadPartCopyRequest;
         }
 
+        private void updateProgressListener() {
```

Comment:
WOuld be great if we could see some comments to describe this logic, with some number examples. 

What I understand is that 10000 is the total size of all parts put together. If say we need to transfer 100 bytes, and the size of a part is 10 bytes, it gives us 10 parts. Scaled to 10000, it is (1/10) * 10000 which is 1000 parts. Is this called eveyrtime we complete the upload of a part ?

Reply:
Can we make byte updates when the upload of a part is in progress ? Like lets say we have 10 parts of 10 bytes each, could we make a progress update in the middle of the 6th parts upload with the number of bytes remaining ? Im guessing it is possible for streaming operations but unsure if TransferProgressUpdate supports this.

Reply:
Added comments to describe logic. This isn't a public method/class so we might not need to be too detailed.

We send the CopyPart request to S3 which handles the operation, so we don't have access to the progress of each individual CopyPart.